api/utils/cfApiClient.js

```diff
@@ -0,0 +1,192 @@
+const request = require('request');
+const url = require('url');
+const CloudFoundryAuthClient = require('./cfAuthClient');
+const config = require('../../config');
+
+class CloudFoundryAPIClient {
+  constructor() {
+    this.authClient = new CloudFoundryAuthClient();
+  }
+
+  createS3ServiceInstance(name, serviceName) {
+    return this.fetchS3ServicePlanGUID(serviceName)
+      .then((servicePlanGuid) => {
+        const body = {
+          name,
+          service_plan_guid: servicePlanGuid,
+          space_guid: this.spaceGUID(),
+        };
+
+        return this.accessToken().then(token => this.request(
+          'POST',
+          '/v2/service_instances?accepts_incomplete=true',
+          token,
+          body
+        ));
+      });
+  }
+
+  createServiceKey(serviceInstanceName, serviceInstanceGuid, keyIdentifier = 'key') {
+    const body = {
+      name: `${serviceInstanceName}-${keyIdentifier}`,
+      service_instance_guid: serviceInstanceGuid,
+    };
+
+    return this.accessToken().then(token => this.request(
+      'POST',
+      '/v2/service_keys',
+      token,
+      body
+    ));
+  }
+
+  createSiteBucket(name, keyIdentifier = 'key', serviceName = 'basic-public') {
+    return this.createS3ServiceInstance(name, serviceName)
+      .then(res => this.createServiceKey(name, res.metadata.guid, keyIdentifier));
+  }
+
+
+  // TODO Check Permissions to Delete Services
```

Comment:
is there an issue or checklist item to come back to this?

Reply:
I'll update the PR